lib/postgres_ext/serializers/active_model/array_serializer.rb

```diff
@@ -145,8 +141,25 @@ def _process_has_many_relation(key, association_reflection, relation_query, ids_
       { table: cte_name, ids_column: id_column_name, foreign_key: association_reflection.foreign_key }
     end
 
-    def _visitor
-      @_visitior ||= object.klass.connection.visitor
+    def _to_sql(arel, bind_values = nil)
+      @_visitor ||= object.klass.connection.visitor
+      if @_visitor.method(:accept).arity == 2
+        collector = Arel::Collectors::SQLString.new
+        collector = @_visitor.accept arel, collector
+        res = collector.value
+        unless bind_values.nil?
+          bind_values.each_with_index do |bv, i|
```

Comment:
The work was actually, dare I say, :scissors: and paste. I'll try to update @crossroads work, but I'm very accepting of a PR from him to help make things more descriptive. :)